rust_src/src/lists.rs

```diff
@@ -236,14 +232,14 @@ impl Iterator for CarIter {
 
 impl From<LispObject> for LispCons {
     fn from(o: LispObject) -> Self {
-        o.as_cons_or_error()
+        o.as_cons().unwrap_or_else(|| wrong_type!(Qconsp, o))
```

Comment:
Would this work as:

    match o.into() {
        Some(cell) => cell,
        None => wrong_type!(Qconsp, o),
    }

Reply:
That would remove one of the final `as_cons` calls.

Reply:
This change and the next one cause a segfault (not sure which one). I'll investigate it later.